src/scipp/_binding.py

```diff
@@ -5,44 +5,79 @@
 import types
 
 from ._scipp import core
-from .utils import get
 
 _dict_likes = [
     core.Dataset, core.Coords, core.Masks, core._BinsMeta, core._BinsCoords,
     core._BinsMasks, core._BinsAttrs
 ]
 
 
+def _get(self, key, default=None):
+    """
+    Return the value for key if key is in present, else default.
+    """
+    try:
+        return self[key]
+    except KeyError:
```

Comment:
Are you using exceptions for control flow here such that this works even if `__contains__` is not available?

Reply:
No. If there is no `__contains__`, it raises `TypeError`. I used an exception to avoid checking `key in self` twice (`self[key]` performs this check).

Reply:
The counter argument would be that exception handling may add even more overhead? But neither should matter in our case.

Reply:
AFAIK, exception handling is as cheap as an `if` in Python and maybe even cheaper. Especially in Python 3.11 where it will be entirely free as long as there is no exception.
But you are right, it shouldn't matter here.